src/components/Form/PhoneField.js

```diff
@@ -0,0 +1,55 @@
+
+import React, {useEffect, useState} from 'react'
+import styled from 'styled-components'
+import { useTranslation } from 'react-i18next'
+import PropTypes from 'prop-types'
+import Country from '../Country'
+import Field from './Field'
+import { useField } from 'formik'
+import { useCallback } from 'react'
+    
+const Wrapper = styled.div`
+
+    & .input-group-prepend{
+        border: 1px solid #ced4da;
+        border-top-left-radius: .25rem;
+        border-bottom-left-radius: .25rem;
+    }
+`
+
+export default function PhoneField({isSubmitting}){
+    const {t} = useTranslation('form')
+        
+    const [lang, setLang] = useState(null)
+    const [,,helper] = useField('country')
+    const {setValue} = helper
+
+    const memoizedSetValueFunc = useCallback(setValue, [])
+
+    useEffect(() => {
+        const {language} = navigator
+        const lang = language.split('-').pop()
+        setLang(lang)
+        memoizedSetValueFunc(lang)
+    }, [memoizedSetValueFunc])
+
+    const onCountrySelect = (value) => {
+        memoizedSetValueFunc(value)
+    }
+
+    return (<Wrapper>
+        <Field
+            prepend={lang && <Country defaultCountry={lang} onSelect={onCountrySelect} />}
```

Comment:
You're right. I did it because `setValue` from formik useField hook triggered an infinite rerender. I had to memoize the function. I fix it. I inject the default country (retrieved from browser lang)  to phone field component